sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/EmptyPartitionReader.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import java.io.IOException
+
+import org.apache.spark.sql.sources.v2.reader.PartitionReader
+
+/**
+ * A [[PartitionReader]] with empty output.
+ */
+class EmptyPartitionReader[T] extends PartitionReader[T] {
+  override def next(): Boolean = false
+
+  override def get(): T =
+    throw new IOException("No records should be returned from EmptyDataReader")
+
+  override def close(): Unit = {}
+}
```

Comment:
This also looks like a good example which is irrelevant to ORC V2 migration.

Reply:
Yep. Please. Your commit will be referenced by many developers.

It's a design issue instead of `size`. The kind of framework classes for DSv2 should be outside of this ORC specific PR (`[SPARK-23817][SQL] Migrate ORC file format read path to data source V2`).

Reply:
how about we change the PR title to `create file format v2 migration infra and migrate ORC`? If we only commit specific code to migrate orc, it's useless and will be refactored right before we migrate the next file format.

Reply:
What about just making another PR containing a subset of this PR?
It's good to demonstrate the framework's feasibility, but it's too big in a single PR.
After merging the framework PR, what we need in this PR is just simple rebasing.

Reply:
@cloud-fan thanks, I have updated the title and description.
@dongjoon-hyun my concern is: if we create a PR for the framework only, then it would be hard to prove it works, even with some dummy data source in the unit test.

Reply:
No, what I suggested is keeping this PR with the framework code.
The framework code PR consists of just a subset of code from here.
And you can give the pointer here. After merging framework PR, you can rebase this PR in order to remove the framework code.

Reply:
To be clear, I'm not suggesting additional test code for the framework PR.

Reply:
We need someting to prove the framework works, I think ORC is a good example, as it has filter pushdown and columnar scan. It may be too wasteful to write a lot of dummy examples in the framework only PR.

Reply:
? Why do we need a new code. My suggestion is making a PR containing the following files, and give a pointer to this PR.
```
DataSourceStrategy.scala
EmptyPartitionReader.scala
FileDataSourceV2.scala
FileDataSourceV2FallBackSuite.scala
FilePartitionReader.scala
FilePartitionReaderFactory.scala
FilePartitionUtil.scala
FileScan.scala
FileScanBuilder.scala
FileScanRDD.scala
FileTable.scala
HiveSessionStateBuilder.scala
PartitionFileUtil.scala
PartitionRecordReader.scala
```

Reply:
If you want, I can make a PR with @gengliangwang 's authorship.

Reply:
I've talked with @cloud-fan . Got it. Please ignore my previous comment. Let's use ORC as an example. Sorry for the delay, @gengliangwang .

Reply:
@cloud-fan @dongjoon-hyun we should not block on such argument. I will create a PR for the framework.

Reply:
@dongjoon-hyun great! I think both approaches are OK. Let's keep it in the current way.